src/Component.php

```diff
@@ -5,12 +5,155 @@
 namespace MyComponent;
 
 use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
+use Keboola\Csv\CsvFile;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+use Symfony\Component\Serializer\Encoder\JsonDecode;
+use Symfony\Component\Serializer\Encoder\JsonEncode;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
 
 class Component extends BaseComponent
 {
+    /**
+     * @var JsonEncode
+     */
+    private $jsonEncode;
+
+    /**
+     * @var JsonDecode
+     */
+    private $jsonDecode;
+
+    /**
+     * @var Filesystem
+     */
+    private $fs;
+
+    private function validateManifest(array $manifest, string $fileName) : void
+    {
+        if (!isset($manifest['columns'])) {
+            throw new UserException(
+                'Manifest file for table ' . $fileName . ' does not specify columns.'
+            );
+        }
+        if (!isset($manifest['delimiter'])) {
+            throw new UserException(
+                'Manifest file for table ' . $fileName . ' does not specify delimiter.'
+            );
+        }
+        if (!isset($manifest['enclosure'])) {
+            throw new UserException(
+                'Manifest file for table ' . $fileName . ' does not specify enclosure.'
+            );
+        }
+    }
+
+    private function getMaxColCount(SplFileInfo $file, string $delimiter, string $enclosure) : int
+    {
+        $csvFile = new CsvFile($file->getPathname(), $delimiter, $enclosure);
+        $maxColCount = 0;
+        foreach ($csvFile as $row) {
+            $maxColCount = max($maxColCount, count($row));
+        }
+        return $maxColCount;
+    }
+
+    private function fillHeader(array $header, int $maxColCount) : array
+    {
+        for ($i = 0; $i < $maxColCount; $i++) {
+            if (empty($header[$i])) {
+                $header[$i] = 'auto_col_' . $i;
+            }
+        }
+        return $header;
+    }
+
+    private function orthogonalize(
+        SplFileInfo $file,
+        string $destination,
+        int $maxColCount,
+        string $delimiter,
+        string $enclosure
+    ) : void {
+        $sourceCsv = new CsvFile($file->getPathname(), $delimiter, $enclosure);
+        $destinationCsv = new CsvFile($destination, $delimiter, $enclosure);
+        foreach ($sourceCsv as $index => $row) {
+            $destinationCsv->writeRow(array_pad($row, $maxColCount, ''));
+        }
+    }
+
+    private function writeManifest(SplFileInfo $file, array $manifest, array $header) : void
+    {
+        $manifest["columns"][] = $header;
```

Comment:
Nemělo by to být `$manifest["columns"] = $header;`? Header je pole sloupců v headeru, ne - `$manifest["columns"] = ['id', 'auto_col_1'];`?

Reply:
ano, samozrejme, 🤦‍♂️